### DISCUSSION TITLE
Version 5 is in progress

### DISCUSSION THREAD

**Discussion by shinsenter** [shinsenter/php#38] [Announcements]
## About the version 5

We are rolling out updated docker images (v5.x) based on the official php images.
https://github.com/shinsenter/php/tree/5.x

Rest assured we have taken steps to minimize the impact of this change on your existing setup.

Your feedback would be most appreciated, so if you could share your `Dockerfile`, `docker-compose.yml` or other customizations, it would help us ensure a smooth transition.

Please let me know if you have any other questions or concerns - we're committed to making this a seamless process.

Best regards.

**Comment by hameno** [comment]
Please don't overwrite existing tags - they should really be immutable otherwise you risk breaking downstream builds (as currently our builds are broken because of recent changes...)
Apparently you switched from ubuntu to debian and now steps using apt-get fail.
Please adjust the naming of your tags to provide immutable tags that never change after first upload...

**Comment by shinsenter** [comment]
@hameno 

I apologize for the changes coming from the new docker images that may affect your current project.

We are using different docker image tags to maintain the old docker image versions. All old docker images have been published with `*-deprecated` tag names. However, we will no longer provide any update for the old docker images.
https://hub.docker.com/r/shinsenter/php/tags?page=1&name=-deprecated

New docker images are built from the official php docker images, ensuring packages receive the latest bug fixes from upstream compared to the old image versions. I hope you can understand and be patient, and we are trying to minimize the impact on projects using the old docker image versions.

Regarding your issue, I understand it may be difficult for you to migrate your containers for production, however I think it should not be too complex if you don't have too many customizations from the base image.

If you don't mind sharing with me the setup for your project, I think I may be able to assist you in migrating your docker container to work with the new docker images.

Best regards

**Comment by shinsenter** [comment]
## Latest Image Tags

Check our [Docker Hub](https://hub.docker.com/r/shinsenter/php/tags) for all available tags.

Docker image tags ending in `-alpine` or `-tidy` indicate Docker images built on the Alpine Linux base OS. These Docker images are lightweight, helping to speed up builds and save bandwidth for your CI/CD.

## Stable Image Tags

We build new Docker images daily to ensure they stay up-to-date with the latest upstream updates for PHP, base OS, Composer, etc. The images in this repo are regularly updated under same tag names.

For stable versions you can depend on in production, we also apply version tags on another repo.

> View Stable Tags: https://hub.docker.com/r/shinsenter/php-archives/tags

This way, you get frequently updated images under static tags, providing both latest code and production stability.
